scaffolding-chef/lib/scaffolding.sh

```diff
@@ -139,6 +139,7 @@ interval = 1800
 splay = 180
 run_lock_timeout = 1800
 log_level = "warn"
+chef_client_ident = "" # this is blank by default so it can be populated from the bind
```

Comment:
🤔 having a default that is unset or blank seems odd to me.
What are the potential pitfalls of setting this to a string like "default"?

Reply:
this bind is expecting a pkg_ident, so if you wanted some kind of false default value, it would need to be in the form of `origin/package/1.2.3/201811...`

I don't believe there's a canonical way to put in an example package ident (ala example.com) that the supervisor would understand. So I think leaving it blank is the best option.

Reply:
Ahh ok, that makes sense. 👍

Reply:
Do bound services _use_ the value of this? Or is it only that the value changed that matters to trigger a rerun of bound services?

Reply:
It's possible to do both, either react solely on the changed value, or actually use it in a configuration file or in the run hook.

In my case I'm using it to provide additional logging for the InSpec service to declare which run it's going against, like so:

`echo "Executing {{pkg.ident}} against {{bind.chef.first.cfg.chef_client_ident}}."`

In either case, we're calling it an `ident` so I would expect consumers to use it as an ident.

Reply:
This can't work for a few reasons:

First, we need to set that pkg_ident AFTER the first converge. Thus, doing it at runtime with hab config apply is necessary.


Even if doing it during the build was possible, it still wouldn't be ideal because the $pkg_ident isn't determined until the end of the build.

It's also a question if scaffolding will even be around in the future, so I'd rather avoid a scaffolding specific implementation.

Reply:
Fair 'nuff. I'm OK with this as-is, then.

As an aside on the details, though ...

> First, we need to set that pkg_ident AFTER the first converge. Thus, doing it at runtime with hab config apply is necessary.

Gotcha. I hadn't meant to say that we should remove the `hab config apply` from the run hook. I understand that a new configuration needs to occur to trigger anything else. Does the trigger-via-bind behavior depend on the value _changing_?

> $pkg_ident isn't determined until the end of the build

I'm not sure about that. `$pkg_ident` ought to be known by this point, especially if `$pkg_prefix` has been computed which seems to be indicated by having been used 5 lines up from this line.

Reply:
Yes, the trigger-via-bind behavior depends on the value changing. The idea is that when new cookbook code is published, and after the converge executes for the first time, that we update the value, and that triggers the run hook via a bind.